lib/poller.js

```diff
@@ -25,19 +25,22 @@ var Poller = exports.Poller = function(config, swfClient) {
    this.config = config ;
    this.pause_poller = false;
    this.logger = config.logger || console;
-   delete config.logger
+   delete config.logger;
    this.taskLimitation = config.taskLimitation || Number.MAX_SAFE_INTEGER;
    delete config.taskLimitation;
    this.taskQueue = async.queue(function(queueFunc, callback){
-    queueFunc(callback);
+     if (!self.pause_poller) {
+       self.poll();
+     }
+     queueFunc(callback);
    }, this.taskLimitation);
-   this.taskQueue.drain = function(){
-    if (this.pause_poller) {
+   this.taskQueue.unsaturated = function(){
+    if (self.pause_poller) {
       self.resume();
     }
    };
    this.taskQueue.saturated = function(){
-    if (!this.pause_poller) {
+    if (!self.pause_poller) {
```

Comment:
but we didn't test that bacause it would only occur when the queue size would reach MAX INTEGER (never).